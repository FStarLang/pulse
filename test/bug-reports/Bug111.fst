module Bug111

#lang-pulse
open Pulse

[@@expect_failure [114]]
ghost
fn test (x : erased int)
  requires emp
  ensures  emp
{
  // let x = reveal x;
  match x {
    0 -> {
     ();
    }
    _ -> {
     ();
    }
  }
}
