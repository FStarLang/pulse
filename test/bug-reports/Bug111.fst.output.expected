>> Got issues: [
* Error 114 at Bug111.fst(13,2-20,3):
  - Could not verify that this match is exhaustive.
  - Type of pattern (Prims.int) does not match type of scrutinee (FStar.Ghost.erased Prims.int)

>>]
